writers/writer.go

```diff
@@ -5,36 +5,52 @@ package writers
 
 import (
 	"fmt"
+	"io/ioutil"
 
+	"github.com/BurntSushi/toml"
 	"github.com/gogo/protobuf/proto"
 	"github.com/mainflux/mainflux"
-	log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/errors"
+	"github.com/mainflux/mainflux/logger"
 	"github.com/mainflux/mainflux/transformers"
 	"github.com/mainflux/mainflux/transformers/senml"
 	nats "github.com/nats-io/nats.go"
 )
 
+var (
+	errOpenConfFile = errors.New("Unable to open configuration file")
+	errParseConfFile = errors.New("Unable to parse configuration file")
+)
+
 type consumer struct {
 	nc          *nats.Conn
-	channels    map[string]bool
 	repo        MessageRepository
 	transformer transformers.Transformer
-	logger      log.Logger
+	logger      logger.Logger
 }
 
 // Start method starts consuming messages received from NATS.
 // This method transforms messages to SenML format before
 // using MessageRepository to store them.
-func Start(nc *nats.Conn, repo MessageRepository, transformer transformers.Transformer, queue string, channels map[string]bool, logger log.Logger) error {
+func Start(nc *nats.Conn, repo MessageRepository, transformer transformers.Transformer, queue string, subjectsCfgPath string, logger logger.Logger) error {
 	c := consumer{
 		nc:          nc,
-		channels:    channels,
 		repo:        repo,
 		transformer: transformer,
 		logger:      logger,
 	}
 
-	_, err := nc.QueueSubscribe(mainflux.InputChannels, queue, c.consume)
+	subjects, err := LoadSubjectsConfig(subjectsCfgPath)
+	if err != nil {
+		logger.Warn(fmt.Sprintf("Failed to load subjects: %s", err))
+	}
+
+	for _, subject := range subjects {
+		_, err := nc.QueueSubscribe(mainflux.InputChannels + subject, queue, c.consume)
```

Comment:
Please, proceed as proposed in the mainflux/docs review. Use the subject here without prefix and move this prefix into the config file. Also, don't forget to use `mainflux.InputChannels` as default value if `LoadSubjectsConfig` fails.

Reply:
@manuio @drasko OK, I understand. When we set filters in `subjects.toml`, it is OK for me to set the full path.
But I have some interrogations about the default value:
- First point, do I need to restore the constant `mainflux.InputChannels` to `channel.>`. to keep as default value? If I did that, I can simply use this value without concatenation.
- Second point, what is the default value in the file `subjects.toml`? Need I restore `["*"]` or we create an empty array?

Reply:
* yes to restore the constant `mainflux.InputChannels`
* default value should be `["channels.>"]`

Reply:
OK, I understand. Personally both versions are OK and I can use both versions. But I still have a preference for the "current version" with relative path for the following reasons:
- As I have understood the operation of Mainflux, all messages are sent on one channel and it is not possible to send messages without `/channels/<channel_id>`. So for me, it is better to set (in a constant) this prefix.
- It is a little bit more complicated but that is more in line with how work Mainflux.
Maybe a good night's sleep will clear this mess :-)

Reply:
@jonathandreyer please use version with full subject in the config file.